crest/Assets/Crest/Crest/Scripts/Shapes/ShapeGerstnerSplineHandling.cs

```diff
@@ -0,0 +1,238 @@
+﻿// Crest Ocean System
+
+// This file is subject to the MIT License as seen in the root of this folder structure (LICENSE)
+
+using Crest.Spline;
+using UnityEngine;
+
+namespace Crest
+{
+    /// <summary>
+    /// Generates mesh suitable for rendering gerstner waves from a spline
+    /// </summary>
+    public static class ShapeGerstnerSplineHandling
+    {
+        public static bool GenerateMeshFromSpline(Spline.Spline spline, Transform transform, int subdivisions, float radius, int smoothingIterations, ref Mesh mesh)
```

Comment:
Just making a note here. If this was able to take an array of points, then we could work with any spline tool that users use (or favour) already. The spline tool you made is great for this purpose so maybe in the future depending on feedback.

Reply:
yeah good point. i explored changing it but ran into a couple of not-nice things and didnt land on something better.

UpdateMesh() can be called with a list of points which sort of ticks this box.

the bigger issue is that our spline is embedded/hardcoded into ShapeGerstner. i think an adaptor pattern could work for this, like timeprovider.

the current stuff could be moved to an adaptor pattern without needing data changes i think, so i dont think we close the door to this option if we merge it as is, so guess its fine for now

Reply:
Sounds good. I'll resolve this one.